src/torrent.cpp

```diff
@@ -5678,61 +5680,81 @@ namespace {
 
 		boost::scoped_ptr<std::vector<boost::uint8_t> > p(j->buffer.priorities);
 
-		if (j->ret == piece_manager::no_error)
+		if (m_file_priority != *p)
 		{
-			TORRENT_ASSERT(!j->error);
 			m_file_priority = *p;
 			update_piece_priorities();
 		}
-		else
-		{
-			// in this case, some file priorities failed to get set
-			TORRENT_ASSERT(j->error);
-			TORRENT_ASSERT(!m_share_mode);
 
-			if (alerts().should_post<file_error_alert>())
-				alerts().emplace_alert<file_error_alert>(j->error.ec
-					, resolve_filename(j->error.file), j->error.operation_str(), get_handle());
+		if (!j->error) return;
 
-			set_error(j->error.ec, j->error.file);
-			pause();
-		}
+		// in this case, some file priorities failed to get set
+		TORRENT_ASSERT(j->error);
+		TORRENT_ASSERT(!m_share_mode);
+
+		if (alerts().should_post<file_error_alert>())
+			alerts().emplace_alert<file_error_alert>(j->error.ec
+				, resolve_filename(j->error.file), j->error.operation_str(), get_handle());
+
+		set_error(j->error.ec, j->error.file);
+		pause();
 	}
 
 	void torrent::prioritize_files(std::vector<int> const& files)
 	{
 		INVARIANT_CHECK;
 
+		if (is_seed()) return;
+
 		// store file priorities until we get the metadata
 		if (!valid_metadata())
 		{
 			m_file_priority.assign(files.begin(), files.end());
 			return;
 		}
 
+		// the vector need to have exactly one element for every file
+		// in the torrent
+		TORRENT_ASSERT(int(files.size()) == m_torrent_file->num_files());
```

Comment:
This assert is incorrect, since `files` vector is allowed to be of any size, because it is going to be "fixed".
The right place for this is line #2730, where `files` should be replaced with `new_priority`.

Reply:
I thought this was a precondition the user had to satisfy though. I think it's reasonable to drop this precondition though. But it only creates a redundant constraint, it's not a more fundamental problem, right?

Reply:
No, it is also mentioned somewhere in comments that all missing file priorities are treated to be 4.